src/UI/Components/Diagram/Context/CanvasProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react";
+import { dia } from "@inmanta/rappid";
+import { Field, InstanceAttributeModel } from "@/Core";
+import { ComposerServiceOrderItem, DictDialogData } from "../interfaces";
+import { CanvasContext } from "./Context";
+
+/**
+ * CanvasProvider component
+ *
+ * This component is a context provider for the Canvas component and it's children.
+ * It maintains the state for various properties related to the canvas and also provides functions to update these states.
+ *
+ * @param {React.PropsWithChildren<unknown>} props - The properties that define the behavior and display of the component.
+ * @param {React.ReactNode} props.children - The children components to be wrapped by the provider.
+ *
+ * @returns {React.FC<React.PropsWithChildren<unknown>>} The CanvasProvider component.
+ */
+export const CanvasProvider: React.FC<React.PropsWithChildren<unknown>> = ({
+  children,
+}) => {
+  const [looseEmbedded, setLooseEmbedded] = useState<Set<string>>(new Set());
+  const [cellToEdit, setCellToEdit] = useState<dia.CellView | null>(null);
+  const [dictToDisplay, setDictToDisplay] = useState<DictDialogData | null>(
+    null,
+  );
+  const [fields, setFields] = useState<Field[]>([]);
+  const [formState, setFormState] = useState<InstanceAttributeModel>({});
+  const [instancesToSend, setInstancesToSend] = useState<
+    Map<string, ComposerServiceOrderItem>
+  >(new Map());
+  const [isDirty, setIsDirty] = useState(false);
+
+  useEffect(() => {
+    if (!isDirty) {
+      setIsDirty(
+        Array.from(instancesToSend).filter(
```

Comment:
Yes, in case of editing instance, we have all entities already there, but unless are changed the status is set to null, as we don't track the state of the whole instance, but just merge it at the end, but now as I'm thinking about it I created task to discuss the idea that could remove that or simplify it